TemplateOperations/Difference/ExceptionTemplates.wl

```diff
@@ -0,0 +1,72 @@
+(* ::Package:: *)
+
+BeginPackage[
+  "CATemplates`TemplateOperations`Difference`ExceptionTemplates`",
+  {
+    "CATemplates`Basic`",
+    "CATemplates`CATemplate`"
+  }
+];
+
+PossibleInvalidSubsets::usage = "bl1";
+SubstitutionRangeVar::usage = "bl1";
+ExpandVar::usage = "bl1";
+ExpandVars::usage = "bl1";
+InvalidSubSets::usage = "bl1";
+ExceptionTemplates::usage= "ExceptionTemplates[t_List, k_Integer:2, r_Real:1] generate all the templates with variable assignments that make the template t with k colors and r range invalid.";
+
+Begin["`Private`"];
+PossibleInvalidSubsets[intemplate_Association] := PossibleInvalidSubsets[intemplate[["rawList"]]];
+PossibleInvalidSubsets[intemplate_List] := Select[intemplate, (Depth[#] > 1) &];
+
+SubstitutionRangeVar[var_, k_Integer : 2] := Range[0, k^Length[TemplateCoreVars[{var}]] - 1];
```

Comment:
What do you think about renaming this to `ExpansionRange`, moving it to `ExpandTemplate` and using it there instead of doing it manually like we do today?